### PR TITLE
#7: code module added

### PR DESCRIPTION
**Pull request by ShashankKasera**
added code module

### CODE REVIEW

**Comment by ShashankKasera** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>No ReleaseNotes.md found in this PR. Please ensure that you add release notes in ReleaseNotes.md file.</td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  No Github issue Id found in PR Title. Please ensure that the PR title conforms to the format: `<Github_issue_Id>: <Title>`.
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>Having a proper description on PR gives more clarity to the reviewer. Please ensure that the PR has a neatly written description explaining the purpose of this change request.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Let's also assign this pull request to someone for review.</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems">Danger Kotlin</a> against 90e27fdab4ad2dd244bc1d5d7d1e3ab9997536ef
</p>

File: Core/src/main/java/com/example/core/extension/Activity.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package com.example.core.extension
+
+import android.app.Activity
+import android.graphics.drawable.Drawable
+import androidx.annotation.ColorRes
+import androidx.annotation.DrawableRes
+import androidx.core.content.ContextCompat
+
+// private var toast: Toast? = null
```
Comment (mayankkasera) [inline Core/src/main/java/com/example/core/extension/Activity.kt:9]: Remove this.

File: Core/src/main/java/com/example/core/extension/Validation.kt [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    !TextUtils.isEmpty(this) && Patterns.EMAIL_ADDRESS.matcher(this).matches()
+
+fun String.isValidPassword(): Boolean {
+    return !TextUtils.isEmpty(this) && length >= 8 && contains(Regex(PASSWORD_REGEX))
+}
+
+fun String.isValidPhoneNumber(): Boolean = length == 10
+
+fun String.isValidDisplayName(): Boolean {
+    //  val regex = "^[\\p{L}0-9_]+$"
```
Comment (mayankkasera) [inline Core/src/main/java/com/example/core/extension/Validation.kt:21]: Remove all unused comments like this.

File: Core/src/main/java/com/example/core/extension/View.kt [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+}
+
+fun View.getImageDrawable(@DrawableRes id: Int): Drawable? = context.getImageDrawable(id)
+
+fun View.getResolvedColor(@ColorRes id: Int): Int = ContextCompat.getColor(context, id)
+
+fun AppCompatEditText.afterTextChanged(afterTextChanged: (String) -> Unit) {
+    this.addTextChangedListener(object : TextWatcher {
+        override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
+            Log.i("ihek", "beforeTextChanged: ")
```
Comment (mayankkasera) [inline Core/src/main/java/com/example/core/extension/View.kt:52]: Add proper tag name for log, "ihek" looking very bed.

File: Core/src/main/java/com/example/core/extension/Activity.kt (hunk 1 above)
Comment (ShashankKasera) [reply Core/src/main/java/com/example/core/extension/Activity.kt]: done

File: Core/src/main/java/com/example/core/extension/Validation.kt (hunk 2 above)
Comment (ShashankKasera) [reply Core/src/main/java/com/example/core/extension/Validation.kt]: done

File: Core/src/main/java/com/example/core/extension/View.kt (hunk 3 above)
Comment (ShashankKasera) [reply Core/src/main/java/com/example/core/extension/View.kt]: done

**Review by mayankkasera — APPROVED** [review]

**Pull request merged**